examples/pytorch/correct_and_smooth/model.py

```diff
@@ -0,0 +1,254 @@
+
+        self.num_layers = num_layers
+        self.alpha = alpha
+    
+    @torch.no_grad()
+    def forward(self, g, labels, mask=None, post_step=lambda y: y.clamp_(0., 1.)):
+        with g.local_scope():
+            if labels.dtype == torch.long:
+                labels = F.one_hot(labels.view(-1)).to(torch.float32)
+            
+            y = labels
+            if mask is not None:
+                y = torch.zeros_like(labels)
+                y[mask] = labels[mask]
+            
+            last = (1 - self.alpha) * y
+            degs = g.in_degrees().float().clamp(min=1)
+            norm = torch.pow(degs, -0.5).to(labels.device).unsqueeze(1)
+
+            for _ in range(self.num_layers):
+                # Assume the graphs to be undirected
+                g.ndata['h'] = y * norm
+                g.update_all(fn.copy_u('h', 'm'), fn.sum('m', 'h'))
+                y = last + self.alpha * g.ndata.pop('h') * norm
+                y = post_step(y)
+                last = (1 - self.alpha) * y
+            
+            return y
+
+
+class CorrectAndSmooth(nn.Module):
+    r"""
+
+    Description
+    -----------
+    Introduced in `Combining Label Propagation and Simple Models Out-performs Graph Neural Networks <https://arxiv.org/abs/2010.13993>`_
+
+    Parameters
+    ----------
+        num_correction_layers: int
+            The number of correct propagations.
+        correction_alpha: float
+            The coefficient of correction.
+        num_smoothing_layers: int
+            The number of smooth propagations.
+        smoothing_alpha: float
+            The coefficient of smoothing.
+        autoscale: bool, optional
+            If set to True, will automatically determine the scaling factor :math:`\sigma`. Default is True.
+        scale: float, optional
+            The scaling factor :math:`\sigma`, in case :obj:`autoscale = False`. Default is 1.
+    """
+    def __init__(self,
+                 num_correction_layers,
+                 correction_alpha,
+                 num_smoothing_layers,
+                 smoothing_alpha,
+                 autoscale=True,
+                 scale=1.):
+        super(CorrectAndSmooth, self).__init__()
+        
+        self.autoscale = autoscale
+        self.scale = scale
+
+        self.prop1 = LabelPropagation(num_correction_layers, correction_alpha)
+        self.prop2 = LabelPropagation(num_smoothing_layers, smoothing_alpha)
+
+    def correct(self, g, y_soft, y_true, mask):
+        with g.local_scope():
+            assert abs(float(y_soft.sum()) / y_soft.size(0) - 1.0) < 1e-2
+            numel = int(mask.sum()) if mask.dtype == torch.bool else mask.size(0)
+            assert y_true.size(0) == numel
+
+            if y_true.dtype == torch.long:
+                y_true = F.one_hot(y_true.view(-1), y_soft.size(-1)).to(y_soft.dtype)
+            
+            error = torch.zeros_like(y_soft)
+            error[mask] = y_true - y_soft[mask]
```

Comment:
Seems they are using both training and validation labels here: https://github.com/CUAI/CorrectAndSmooth/blob/b910314a59270984f5e249462ee3faa815fc9a0c/outcome_correlation.py#L192